### ISSUE TITLE
Can't use scrcpy

### ISSUE THREAD

**Issue by edent** [fakemanoan/LineageOS-Releases#23]
This is a weird issue. I can't run `scrcpy` - it gives the error "ART APEX data files are untrusted."

Only happens on my fltexx - doesn't happen on any other Lineage device.

Any idea what's causing it?

https://github.com/Genymobile/scrcpy/issues/3467

**Comment by fragtion** [comment]
Same problem when trying to use Vysor, btw. So it's likely a LineageOS issue, not any of these programs?

03-25 15:48:20.297  3616  3692 I RoleService: Granting default roles...
03-25 15:48:20.350  7225  7225 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 0 <<<<<<
03-25 15:48:20.355  7225  7225 I AndroidRuntime: Using default boot image
03-25 15:48:20.356  7225  7225 I AndroidRuntime: Leaving lock profiling enabled
03-25 15:48:20.360  7225  7225 W app_process64: ART APEX data files are untrusted.
03-25 15:48:20.360  7225  7225 E app_process64: Could not open boot oat file for extracting boot class path: File /apex/com.android.art/javalib/boot.art:/system/framework/boot-framework.art!/system/etc/arm64/boot-image.vdex does not exist.
03-25 15:48:20.360  7225  7225 E AndroidRuntime: JNI_CreateJavaVM failed
03-25 15:48:20.421  7228  7228 D ProcessState: Binder ioctl to enable oneway spam detection failed: Invalid argument
03-25 15:48:20.438     0     0 I [5:            cmd: 7228] binder: 7228:7228 ioctl 40046210 7fec297f94 returned -22
03-25 15:48:20.523     0     0 I         : [0:  kworker/u16:6: 3493] vfsspi_work_func_debug ldo:3, sleep:0, irq:0, tz:1, type:viper, cnt_irq:0, adm: 0
03-25 15:48:20.649  7237  7237 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 0 <<<<<<
03-25 15:48:20.655  7237  7237 I AndroidRuntime: Using default boot image
03-25 15:48:20.655  7237  7237 I AndroidRuntime: Leaving lock profiling enabled
03-25 15:48:20.659  7237  7237 W app_process64: ART APEX data files are untrusted.
03-25 15:48:20.659  7237  7237 E app_process64: Could not open boot oat file for extracting boot class path: File /apex/com.android.art/javalib/boot.art:/system/framework/boot-framework.art!/system/etc/arm64/boot-image.vdex does not exist.
03-25 15:48:20.659  7237  7237 E AndroidRuntime: JNI_CreateJavaVM failed

**Comment by edent** [comment]
Still an issue on 20230416.

**Comment by fakemanoan** [comment]
Issue will be resolved in latest upcoming Lineage 20 build

**Comment by fwiep** [comment]
> Issue will be resolved in latest upcoming Lineage 20 build

That is good news, but could you please elaborate on this victory? Was there a recent change on the LineageOS side, or did you alter your own build procedure? Does this fix also apply for LineageOS 19 custom builds?

I would really like to be able to use `scrcpy` once again with my Android 12 devices; your comment is the first shred of hope in a *long* time.

Thanks,
FWieP

**Comment by fakemanoan** [comment]
I noticed randomly when doing other debugging with some art changes I made that when adb had root, scrcpy started to work (rooted debugging in dev options).

It seems to be a usb.rc file issue relating to stopping adbd (i think?). Commit that fixed the issue: https://github.com/samsungexynos7420/android_device_samsung_universal7420-common/commit/f367a42be4957e12931392868d1aeea041694a82

Hope this helps.
